From 7f4a2848e01ddaa3250d9d17e2e87a090b29a1c8 Mon Sep 17 00:00:00 2001
From: Jeffrey Altman <jaltman@auristor.com>
Date: Wed, 16 Nov 2022 17:28:21 -0500
Subject: [PATCH 1/3] lib/krb5: krb5_pac_parse mem leak if pac_header_size
 failure

48 byte memory leak from krb5_pac_parse() each time pac_header_size()
fails.

(cherry picked from commit 02f12fc746341f54a514e9e17bc7d315b91129e8)
Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>
---
 lib/krb5/pac.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/lib/krb5/pac.c b/lib/krb5/pac.c
index cf051cd1f..81eba84e2 100644
--- a/lib/krb5/pac.c
+++ b/lib/krb5/pac.c
@@ -204,9 +204,8 @@ krb5_pac_parse(krb5_context context, const void *ptr, size_t len,
     }
 
     ret = pac_header_size(context, tmp, &header_end);
-    if (ret) {
-	return ret;
-    }
+    if (ret)
+	goto out;
 
     p->pac = calloc(1, header_end);
     if (p->pac == NULL) {
-- 
2.42.0

